### PR TITLE
Update deprecated VSCode settings

### PR DESCRIPTION
**Pull request by CosminGGeorgescu**
### Pull Request Overview

This pull request removes the `rust-client.channel` field, part of the [Rust VSCode extension](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust), which is deprecated as of 15.11.2022, and replaces it with the equivalent setting for the new [rust-analyzer extension](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer). However, there could be a case made for developers that install the extension by source code (somehow ? I'm not familiar with this way), as the [Github repo](https://github.com/rust-lang/vscode-rust) is still public, just archived.

The second field `"rust-analyzer.check.allTargets": false,` was added to fix the `found duplicate lang item ``panic_impl`` ` error and associated squiggles sometimes raised by rust-analyzer, as it runs `cargo check` with the `--all-targets` flag by default and raises an error for each `pub unsafe extern "C" fn panic_fmt(pi: &PanicInfo)` that is not preceded by `#[cfg(not(test))]`.

### Testing Strategy

This pull request was tested by checking with the official rust-analyzer documentation and restarting the rust-analyzer server to see that the added changes had the desired effects.

### TODO or Help Wanted

If there are any suggestion as what more to be enforced upon the VSCode developer, let me know and I'll add them here.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by ppannuto — DISMISSED** [review]
I don't believe anyone on the core team (except maybe @hudson-ayers) use VSCode, so this has always been something we defer to community folks to maintain. Changes look reasonable to me — thanks!

File: .vscode/launch.json
Comment (bradjc) [inline .vscode/launch.json:33]: ?

File: .vscode/settings.json
Comment (bradjc) [inline .vscode/settings.json:7]: ?

Comment (bradjc) [reply .vscode/settings.json:7]: (why remove newline?)

Comment (CosminGGeorgescu) [reply .vscode/settings.json:7]: Was a case of `"files.insertFinalNewline": false` being the default in VSCode and `formatOnSave` just doing it's job. Should I add them back ? Additionally, I could add `"files.insertFinalNewline": true` to `settings.json` and override the user's default, to keep a rule.

File: .vscode/launch.json
Comment (CosminGGeorgescu) [reply .vscode/launch.json:33]: Same as below.

**Comment by hudson-ayers** [comment]
> I don't believe anyone on the core team (except maybe @hudson-ayers) use VSCode, so this has always been something we defer to community folks to maintain. Changes look reasonable to me — thanks!

Nope, I use neovim! So no stake in this, but the changes look fine to me.

**Comment by alexandruradovici** [comment]
I am using VS Code, I'll take a look at the changes, but so far they look good to me.

**Review by bradjc — APPROVED** [review]

**Pull request merged**
